### PR TITLE
Admin request page

### LINKED ISSUE #14: [3pts] Create Requests page for admin
- Create Request routes for admin
- Admin can approve or deny a request

### PR DESCRIPTION
**Pull request by EidHachem**
## In this PR I added the following functionalities:

- Create two links inside the dashboard to list all requests, or to manage the requests.
- In the pending reservation tab, admins can confirm/reject pending requests.
- In the manage reservation tab, admins can list all reservation requests.

closes https://github.com/ichala/short-let-frontend/issues/14

### CODE REVIEW

**Review by d2ndjim — APPROVED** [review]
@EidHachem, Nice work. 
### APPROVED ✔️ 
After fixing the rendering issue

**Review by LYANGEND — COMMENTED** [review]
Goodwork @EidHachem 🥇 

APPROVED ✔️

**Review by ichala — APPROVED** [review]
✔️ Approved

**Review by ichala — APPROVED** [review]
✔️ Approved good job

File: src/Pages/Users/MyReservations.jsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ const MyReservations = () => {
 
   useEffect(() => {
     GetReservationStats(setError, setReservations);
-  }, [reservation, reservations]);
+  }, [reservation, setReservations]);
```
Comment (aimalamiri) [inline src/Pages/Users/MyReservations.jsx]: I'm confused about how you listen to the method change which is never going to be changed. `setReservations` has to mutate the `reservations`, and it's not going to be changed itself, am I right?

**Review by aimalamiri — CHANGES_REQUESTED** [review]
Hi @EidHachem, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

AlmostThereReachGIF

## Highlights


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/Pages/Users/MyReservations.jsx (hunk 1 above)
Comment (EidHachem) [reply src/Pages/Users/MyReservations.jsx]: Hi @aimalamiri you are right, setReservations is a setter function that is guaranteed to be stable by useState, I must have added it by mistake to remove eslint error before passing it to API request function. I will remove it now.

**Review by aimalamiri — APPROVED** [review]
Hi @EidHachem, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

ApprovedDavieGIF


## Highlights

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**